pkg/runtime/runtime.go

```diff
@@ -1179,7 +1180,13 @@ func (a *DaprRuntime) publishMessageGRPC(msg *pubsub.NewMessage) error {
 	}
 
 	if data, ok := cloudEvent[pubsub.DataField]; ok && data != nil {
-		envelope.Data = []byte(cloudEvent[pubsub.DataField].(string))
+		envelope.Data = nil
+
+		if contrib_contenttype.IsStringContentType(envelope.DataContentType) {
+			envelope.Data = []byte(data.(string))
+		} else if contrib_contenttype.IsJSONContentType(envelope.DataContentType) {
```

Comment:
Check for JSON first because theoretically JSON is  StringContentType too. The checks should be more specific first.